### DISCUSSION TITLE
SnowOwl & AWS ElasticSearch

### DISCUSSION THREAD

**Discussion by martingall87** [b2ihealthcare/snow-owl#1033] [Q&A] [answered]
Hello B2i!

At the moment I'm looking to setup a little demo environment with Snow Owl connecting to an OpenSearch (Elastic 7.10) instance in AWS.
 
I wondered if there was any guidance / best practice for connecting SnowOwl successfully to an AWS ElasticSearch instance? I believe AWS ElasticSearch requires all comms to be signed using an AWS v4 sig, however it looks like HTTP Basic Auth can be used instead? 

Any guidance you can provide would be much appreciated,

Many thanks,

Martin

**Comment by cmark** [comment, marked as answer]
Hi @martingall87,

Thank you for raising your question. 
With the most recent releases of Snow Owl 7 and 8 I'm afraid that is not possible anymore.
The last version that supports the AWS OpenSearch Service (which is based on the equivalent of Elasticsearch 7.10.x) is Snow Owl 7.17.6, see the official release version here:
https://github.com/b2ihealthcare/snow-owl/releases/tag/v7.17.6

The official docker image for that version can be found [here](https://hub.docker.com/layers/snow-owl-oss/b2ihealthcare/snow-owl-oss/7.17.6/images/sha256-336990ce06d880873451591aaa0ab375aff9754282dd4aadaae5ea16d3fe46ea?context=explore).

Just an FYI, the main reason for dropping support for Elasticsearch 7.10 was the Log4Shell vulnerability found in the log4j2 jars, which are present in these very versions. It might not affect OpenSearch, but if you are planning to use Elasticsearch 7.10 in production that might be of interest to you. See the original issue [here](https://github.com/b2ihealthcare/snow-owl/issues/964)

If you are interested and would like to test it, there is an experimental [opensearch-support](https://github.com/b2ihealthcare/snow-owl/tree/experiment/opensearch-support) branch available. No official build is available from this branch yet.

> I wondered if there was any guidance / best practice for connecting SnowOwl successfully to an AWS ElasticSearch instance? I believe AWS ElasticSearch requires all comms to be signed using an AWS v4 sig, however it looks like HTTP Basic Auth can be used instead?

In theory, Basic Auth should be enough to get you started, locally for sure, but I think using the current AWS offering as well. There is no official guide out there because here at B2i we only use Elastic stack products and services. If you are willing to share something back with the community, PRs are very much welcome. 😃 

Feel free to contact us if you need further information or help.

Cheers,
Mark

PS. I have converted this issue to a GitHub discussion where you can mark this as an answer if you'd like. That might help future adventurers when they seek knowledge. 🧙
